lib/array.js

```diff
@@ -38,6 +38,60 @@ const map = (
   }
 };
 
+const NS_PER_SEC = 1e7;
+
+const asyncMap = (
+  // Non blocking map
+  items, // array, incoming
+  fn, // function, (value, index) => {}
+  {
+    min = 5,
+    max = 30,
+  } = {}
+) => {
+  let count = 0;
+  let setTimeoutCount = 0;
+
+  let maxTime = 0;
+  let minTime = Infinity;
+  let sumTime = 0;
+
+  let begin = process.hrtime();
+
+  const halfRange = (max - min) / 2;
+  const middle = min + halfRange;
+
+  let iterMax = middle;
+
+  const countNecessaryNumber = () => {
+    let time = process.hrtime(begin);
+    time = time[0] + NS_PER_SEC + time[1];
```

Comment:
Using addition to the `NS_PER_SEC` here seems to be a typo, it should probably be multiplication:
```suggestion
    time = time[0] * NS_PER_SEC + time[1];
```